open_schools_platform/history_management/views.py

```diff
@@ -0,0 +1,87 @@
+from rest_framework.views import APIView
+from rest_framework.response import Response
+from drf_yasg.utils import swagger_auto_schema
+
+from open_schools_platform.api.swagger_tags import SwaggerTags
+from open_schools_platform.common.views import swagger_dict_response
+from open_schools_platform.user_management.users.selectors import get_user
+from open_schools_platform.organization_management.organizations.selectors import get_organization
+from open_schools_platform.organization_management.employees.selectors import get_employee
+from .serializers.user_serializer import UserHistorySerializer
+from .serializers.organization_serializer import OrganizationHistorySerializer
+from .serializers.employee_serializer import EmployeeHistorySerializer
+from .serializers.circle_serializer import CircleHistorySerializer
+from .serializers.student_serializer import StudentHistorySerializer
+
+from ..organization_management.circles.selectors import get_circle
+from ..student_management.students.selectors import get_student
+
+
+class UserHistoryApi(APIView):
+    @swagger_auto_schema(
+        operation_description="Get user history.",
+        tags=[SwaggerTags.HISTORY_MANAGEMENT],
+        responses={200: swagger_dict_response({'results': UserHistorySerializer(many=True)}),
+                   404: "There is no such user"},
+    )
+    def get(self, request, pk):
+        user = get_user(filters={"id": pk},
+                        empty_exception=True,
+                        empty_message="There is no such user")
```

Comment:
user=request.user!!!!!!!!!

Reply:
No! We should get user by pk, not current one that have sent the request.

Reply:
It is not secure in my opinion. Anyone can see personal data like ip, last_login_time and etc